src/safeposix/shm.rs

```diff
@@ -0,0 +1,97 @@
+// Filesystem metadata struct
+#![allow(dead_code)]
+
+use crate::interface;
+use super::syscalls::fs_constants::*;
+use super::syscalls::sys_constants::*;
+
+use super::cage::Cage;
+
+pub static SHM_METADATA: interface::RustLazyGlobal<interface::RustRfc<ShmMetadata>> = 
+    interface::RustLazyGlobal::new(|| interface::RustRfc::new(ShmMetadata::init_shm_metadata())); 
+
+pub struct ShmSegment {
+    pub shminfo: interface::ShmidsStruct,
+    pub key: i32,
+    pub size: usize,
+    pub filebacking: interface::ShmFile,
+    pub rmid: bool
+}
+
+pub fn new_shm_segment(key: i32, size: usize, cageid: i32, uid: u32, gid: u32, mode: u16) -> ShmSegment {
+    ShmSegment::new(key, size, cageid, uid, gid, mode)
+}
+
+impl ShmSegment {
+    pub fn new(key: i32, size: usize, cageid: i32, uid: u32, gid: u32, mode: u16) -> ShmSegment {
+        let filebacking = interface::new_shm_backing(key, size).unwrap();
+
+        let time = interface::timestamp() as isize; //We do a real timestamp now
+        let permstruct = interface::IpcPermStruct { __key: key, uid: uid, gid: gid, cuid: uid, cgid: gid, mode: mode, __seq: 0 };
```

Comment:
Why are there underlines before key and seq?

Reply:
That's how they're referenced in the linux header file, so I'm keeping it consistent.